test/common/cache_utils.py

```diff
@@ -0,0 +1,84 @@
+import os
+import json
+import torchtext
+import functools
+import unittest
+from .parameterized_utils import load_params
+
+CACHE_STATUS_FILE = '.data/cache_status_file.json'
+
+
+def check_cache_status(_func=None, *, input_dataset_name=None, input_split=None):
```

Comment:
Can't these verifications be run once at import time instead of for each test separately again and again? The wrapper can then simply store the information that if the test is run it must fail, else just runs the test.

Reply:
Yes, now we are checking the cache status in setUpClass method, and raise exception if cache was not successfully build for all the raw datasets. This is a strict policy where we let the whole test suit fail if at-least one dataset is missing in the cache.